src/viewProviders/flinkArtifacts.ts

```diff
@@ -52,13 +57,25 @@ export class FlinkArtifactsViewProvider
 
     this._onDidChangeTreeData.fire();
   }
+
   getTreeItem(element: FlinkArtifact): TreeItem {
     return new FlinkArtifactTreeItem(element);
   }
 
   get computePool(): CCloudFlinkComputePool | null {
     return this.resource;
   }
+
+  protected setEventListeners(): import("vscode").Disposable[] {
+    return [
+      ...super.setEventListeners(),
+      udfUploadCompleted.event(this.udfUploadCompletedHandler.bind(this)),
+    ];
```

Comment:
No need to override `setEventListeners()` since the base class supports custom listeners per subclass:
https://github.com/confluentinc/vscode/blob/341677b465226314dc1acefec9666b99c55ef839/src/viewProviders/baseModels/base.ts#L126-L129

Reply:
I assumed this meant remove the spread operator line, but when I did, I lost the ability to set a compute pool on artifacts (nothing happened when I selected one). added back in for now

Reply:
Ah, no I mean using `setCustomEventListeners` instead of `setEventListeners` so it keeps the base class behavior - like this:
```ts
protected setCustomEventListeners(): Disposable[] {
  return [
    udfUploadCompleted.event(this.udfUploadCompletedHandler.bind(this)),
  ];
}
```